### PR TITLE
Expose setting a Window Icon

### PR DESCRIPTION
**Pull request by thewizzy**
The ability to set a window icon is available in floem-winit but it isn't exposed through floem. This allows the floem dev to set a window icon as part of the window configuration.

There is an example included and it borrows heavily from the [floem winit window_icon example](https://github.com/lapce/winit/blob/master/examples/window_icon.rs) - including the same icon file.

### CODE REVIEW

File: src/window.rs [hunk 1]
```diff
@@ -17,6 +18,7 @@ pub struct WindowConfig {
     pub(crate) transparent: Option<bool>,
     pub(crate) fullscreen: Option<Fullscreen>,
     pub(crate) window_icon: Option<bool>,
+    pub(crate) window_icon_value: Option<Icon>,
```
Comment (thewizzy) [inline src/window.rs:21]: I think we could change the previous line `window_icon` to be an Option<Icon> instead and remove `window_icon_value`.
This is a breaking change to the API - so the choice is for floem devs.

Comment (dzhou121) [reply src/window.rs:21]: I think we can re-use `window_icon` because it's not used anywhere

**Comment by panekj** [comment]
It should be noted from where the icon comes from and what licence it is under

**Comment by thewizzy** [comment]
> It should be noted from where the icon comes from and what licence it is under or the icon should be replaced with one that doesn't require attribution/licence (for example Ferris could be used like in #116)

Absolutely. I have updated it to use a copy of existing assets that you mentioned. Thank you.

File: src/window.rs (hunk 1 above)
Comment (thewizzy) [reply src/window.rs]: This has been changed. It is much nicer this way, thank you for the agreement.

**Pull request merged**